### PR TITLE
412 grouping uid

### PR DESCRIPTION
**Pull request by ShiriMoran**
solves #412

### CODE REVIEW

File: pkg/ibmvpc/analysis_output_test.go [hunk 1]
```diff
@@ -515,6 +515,13 @@ var tests = []*vpcGeneralTest{
 		grouping:    true,
 		format:      vpcmodel.Text,
 	},
+	// grouping test of identical names different names that should not be grouped
```
Comment (adisos) [inline pkg/ibmvpc/analysis_output_test.go:518]: what do you mean by `identical names different names` ?

File: pkg/ibmvpc/examples/out/analysis_out/tgw_basic_example_dup_subnet_names_all_vpcs_subnetsBased_withPGW_with_grouping.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Connectivity between VPCs connected by TGW local-tg-ky (UID: crn:161)
+ky-vpc1-net1 => ky-vpc1-net1 : All Connections
+ky-vpc1-net1 => ky-vpc1-net1 : All Connections
```
Comment (adisos) [inline pkg/ibmvpc/examples/out/analysis_out/tgw_basic_example_dup_subnet_names_all_vpcs_subnetsBased_withPGW_with_grouping.txt:3]: open an issue with reference to this example output, where subnet name should also include vpc , at least in the cross-vpc connectivity ?

Comment (ShiriMoran) [reply pkg/ibmvpc/examples/out/analysis_out/tgw_basic_example_dup_subnet_names_all_vpcs_subnetsBased_withPGW_with_grouping.txt:3]: Sure.  https://github.com/np-guard/vpc-network-config-analyzer/issues/498
But note that we can also have two subnets with the same name in the same VPC (though one may claim in is less likely); As I have used this input file to demonstrate the dup names issue I could have chosen another one with only one VPC.
If we really want to tackle dup names we should perhaps issue some lint error. Or have some renaming mechanism? perhaps open issue for this as well?

Comment (adisos) [reply pkg/ibmvpc/examples/out/analysis_out/tgw_basic_example_dup_subnet_names_all_vpcs_subnetsBased_withPGW_with_grouping.txt:3]: ok, updated the issue to contain these as two related tasks

File: pkg/ibmvpc/analysis_output_test.go [hunk 3]
```diff
@@ -515,6 +515,13 @@ var tests = []*vpcGeneralTest{
 		grouping:    true,
 		format:      vpcmodel.Text,
 	},
+	// grouping test of identical names different resources and thus different UIDs that should not be grouped
+	{
+		inputConfig: "tgw_basic_example_dup_subnet_names",
+		useCases:    []vpcmodel.OutputUseCase{vpcmodel.AllSubnets},
+		grouping:    true,
+		format:      vpcmodel.Text,
+	},
```
Comment (adisos) [inline pkg/ibmvpc/analysis_output_test.go:524]: Is this test supposed to fail before the change in the grouping code? Isn't the grouping prevented for cross-vpc subnets anyway?

Comment (ShiriMoran) [reply pkg/ibmvpc/analysis_output_test.go:524]: No, grouping is not prevented in cross-vpc subnets:
E.g.:
```
Connectivity between VPCs connected by TGW local-tg-ky (UID: crn:578)
subnet11-ky,subnet12-ky => subnet21-ky : All Connections
subnet21-ky => subnet11-ky,subnet12-ky : All Connections
subnet21-ky => subnet31-ky,subnet32-ky : All Connections
subnet31-ky => subnet11-ky,subnet12-ky : All Connections *
subnet31-ky,subnet32-ky => subnet21-ky : All Connections
```

Comment (ShiriMoran) [reply pkg/ibmvpc/analysis_output_test.go]: Done, I also verified that the test fails in main.

Btw, this issue is relevant only to subnets with the same name within the same vpc. Perhaps we should have given it lower priority. Anyways now its fixed.

**Review by adisos — APPROVED** [review]

**Pull request merged**